src/Sil/Component/Contact/Tests/Unit/ContactTest.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+/*
+ * Copyright (C) 2015-2017 Libre Informatique
+ *
+ * This file is licenced under the GNU LGPL v3.
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Sil\Component\Contact\Tests\Unit;
+
+use PHPUnit\Framework\TestCase;
+use Sil\Component\Contact\Model\Contact;
+use Sil\Component\Contact\Model\Address;
+use Sil\Component\Contact\Model\Phone;
+use Sil\Component\Contact\Model\City;
+
+/**
+ * @author Romain Sanchez <romain.sanchez@libre-informatique.fr>
+ *
+ * @coversDefaultClass \Sil\Component\Contact\Model\Contact
+ */
+class ContactTest extends TestCase
+{
+    /**
+     * The first address added to the contact should be set as the default address.
+     */
+    public function testSetDefaultAddressWhenAddingFirstAddress()
+    {
+        $contact = new Contact();
+        $address = new Address('foo street', new City('foo city', '55000'), 'foo country');
+
+        $contact->addAddress($address);
+
+        $this->assertNotNull($contact->getDefaultAddress());
+        $this->assertSame($address, $contact->getDefaultAddress());
+    }
+
+    /**
+     * An exception should be thrown when adding an address
+     * that is already associated to this contact.
+     */
+    public function testAddAlreadyExistingAddress()
+    {
+        $contact = new Contact();
+        $address = new Address('foo street', new City('foo city', '55000'), 'foo country');
+
+        $contact->addAddress($address);
+
+        $this->expectException(\InvalidArgumentException::class);
+
+        $contact->addAddress($address);
+    }
+
+    /**
+     * An exception should be thrown when trying to remove the only address of a contact.
+     */
+    public function testRemoveOnlyAddress()
+    {
+        $contact = new Contact();
+        $address = new Address('foo street', new City('foo city', '55000'), 'foo country');
+
+        $contact->addAddress($address);
+
+        $this->expectException(\InvalidArgumentException::class);
+
+        $contact->removeAddress($address);
+    }
+
+    /**
+     * An exception should be thrown when trying to remove an address from a contact with no addresses.
+     */
+    public function testRemoveWhenNoAddress()
```

Comment:
Remove this test, the test `testRemoveNonExistingAddress()` check the same things...